### PR TITLE
ICU-22583 BRS74: Version number update for 74.2

### PR DESCRIPTION
**Pull request by rp9-next**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22583
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by markusicu — DISMISSED** [review]
lgtm tnx!

**Comment by mihnita** [comment]
The java versions (all the `pom.xml`) files should use a `-SNAPSHOT` version. So `7.2-SNAPSHOT`

**Comment by markusicu** [comment]
> @rp9-next Please resolve the merge conflict; I can then re-approve.

The conflict is in .../icuver.txt, so the Java .../icudt74b/icuver.res will need to be updated too.
FYI @mihnita

File: icu4j/releases_tools/shared.sh [hunk 1]
```diff
@@ -5,8 +5,8 @@
 export MAVEN_ARGS='--no-transfer-progress'
 
 # Version update!
-export artifact_version='74.1-SNAPSHOT'
-export github_rel_version='74rc'
+export artifact_version='74.2'
+export github_rel_version='74_2'
```
[the remark this answers is not in the document]
Comment (mihnita) [reply icu4j/releases_tools/shared.sh:9]: No big deal.
That's a script I prepared to build and zip all the release files, including zipped complete javadoc and all.
But the release was done by Yoshito "by hand"

**Review by mihnita — APPROVED** [review]
Thank you, lgtm

**Review by echeran — APPROVED** [review]
rslgtm

**Review by markusicu — CHANGES_REQUESTED** [review]
Thanks for updating icuver.txt, but as I said before, that's incomplete without also updating the corresponding icuver.res in the Java tree.

**Comment by mihnita** [comment]
Submitted as PR https://github.com/unicode-org/icu/pull/2734

**Pull request closed without merging**